pkg/kubectl/cmd/scale.go

```diff
@@ -110,13 +110,24 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 	}
 
 	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
+	if len(resourceVersion) != 0 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --resource-version with multiple controllers")
+	}
 	currentSize := cmdutil.GetFlagInt(cmd, "current-replicas")
+	if currentSize != -1 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --current-replicas with multiple controllers")
+	}
 	precondition := &kubectl.ScalePrecondition{currentSize, resourceVersion}
 	retry := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
-	waitForReplicas := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
```

Comment:
I don't object to changing the waiting behavior, but we do need to ensure this doesn't break e2e.

cc @bprashanth in case he's aware of any problems this might cause.